src/python/pants/rules/core/binary.py

```diff
@@ -32,12 +35,25 @@ class CreatedBinary:
 
 
 @console_rule
-async def create_binary(addresses: BuildFileAddresses, console: Console, workspace: Workspace, options: Binary.Options) -> Binary:
+async def create_binary(
+    addresses: BuildFileAddresses,
+    console: Console,
+    workspace: Workspace,
+    options: Binary.Options,
+    options_bootstrapper: OptionsBootstrapper,
+    ) -> Binary:
   with Binary.line_oriented(options, console) as print_stdout:
-    print_stdout("Generating binaries in `dist/`")
+    global_options = options_bootstrapper.bootstrap_options.for_global_scope()
+    pants_distdir = global_options.pants_distdir
+    relpath = os.path.relpath(pants_distdir, get_buildroot())
```

Comment:
This should instead request the BuildRoot singleton. Rip grep `BuildRoot` and I think we have a rule that uses it (on phone, sorry I’m not finding it)

Reply:
Also pathlib.Path preferred

Reply:
is there a way to do what `os.path.relpath` is doing using `pathlib`? https://docs.python.org/3.7/library/pathlib.html#correspondence-to-tools-in-the-os-module says there isn't an exact equivalent.